client/packages/invoices/src/InboundShipment/ListView/AppBarButtons.tsx

```diff
@@ -95,35 +104,25 @@ export const AppBarButtons = ({
           label={t('button.export')}
         />
       </Grid>
-
-      {showManuallyLinkModal && (
-        <LinkInternalOrderModal
-          requestRequisitions={data?.nodes}
-          isOpen={linkRequestModalController.isOn}
-          onClose={linkRequestModalController.toggleOff}
-          onRowClick={row => {
-            createInvoice(name?.id ?? '', row.id);
-            linkRequestModalController.toggleOff();
-          }}
-          isLoading={internalOrderIsLoading}
-          onNextClick={() => {
-            if (name) {
-              createInvoice(name.id);
-            }
-          }}
-        />
-      )}
+      <LinkInternalOrderModal
+        isOpen={linkRequestModalController.isOn}
+        onClose={linkRequestModalController.toggleOff}
+        onRowClick={row => {
+          createInvoice(name?.id ?? '', row.id);
+          linkRequestModalController.toggleOff();
+        }}
+        onNextClick={() => {
+          if (name) {
+            createInvoice(name.id);
+          }
+        }}
+        name={name}
+      />
       <SupplierSearchModal
         open={invoiceModalController.isOn}
         onClose={invoiceModalController.toggleOff}
         onChange={async nameRow => {
-          setName(nameRow);
```

Comment:
This was causing some weird behaviour because the action would be executed (ie check if there are internal orders) before data can be refetched. 

So if a user had previous created a manual inbound shipment from store B which had no internal orders to store A, and then went to create a manual inbound shipment from store C which DID have internal orders to store A, it would still use the cached data from store B. So needed to be refetched.